camel-core/src/main/docs/validator.adoc

```diff
@@ -0,0 +1,190 @@
+| Name | Description
+
+| ref | Reference to the link:endpoint.html[Endpoint] ID
+
+| uri | link:endpoint.html[Endpoint] URI
+|=======================================================================
+Here is an example to specify endpoint URI in Java DSL:
+[source,java]
+-------------------------------------------------------------------
+validator()
+    .type("xml")
+    .withUri("validator:xsd/schema.xsd");
+-------------------------------------------------------------------
+
+And here is an example to specify endpoint ref in XML DSL:
+[source,xml]
+-------------------------------------------------------------------
+<endpointValidator uri="validator:xsd/schema.xsd" type="xml"/>
+-------------------------------------------------------------------
+
+
+
+[[Validator-Custom]]
+Custom Validator Options
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Note that Validator must be a subclass of `org.apache.camel.spi.Validator`
+|=======================================================================
+| Name | Description
+
+| ref | Reference to the custom Validator bean ID
+
+| className | Fully qualified class name of the custom Validator class
+|=======================================================================
+Here is an example to specify custom Validator class:
+Java DSL:
+[source,java]
+-------------------------------------------------------------------
+validator()
+    .type("json")
+    .withJava(com.example.MyCustomValidator.class);
+-------------------------------------------------------------------
+
+XML DSL:
+[source,xml]
+-------------------------------------------------------------------
+<customTransformer className="com.example.MyCustomValidator" type="json"/>
+-------------------------------------------------------------------
+
+
+
+[[Validator-Examples]]
+Examples
+^^^^^^^
+
+For example to declare the Endpoint Validator which uses
+validator component to validate `xml:ABCOrder`, we can do as follows:
+
+Java DSL:
+[source,java]
+-------------------------------------------------------------------
+validator()
+    .type("xml:ABCOrder")
+    .withUri("validator:xsd/schema.xsd");
+-------------------------------------------------------------------
+
+XML DSL:
+[source,xml]
+-------------------------------------------------------------------
+<camelContext id="camel" xmlns="http://camel.apache.org/schema/spring">
+    <validators>
+        <endpointValidator uri="validator:xsd/schema.xsd" type="xml:ABCOrder"/>
+    </validators>
+    ....
+</camelContext>
+-------------------------------------------------------------------
+
+If you have following route definition, above validator will be applied when `direct:abc` endpoint
+receives the message. Note that the `validate` attribute on the inputType declaration is set to `true`:
+
+Java DSL:
+[source,java]
+-------------------------------------------------------------------
+from("direct:abc")
+    .inputType("xml:ABCOrder", true)
```

Comment:
Thanks for the good catches Claus! especially the `inputTypeWithValidate` is much more descriptive in Java DSL. I'll fix those soon.